azure-mgmt-graph-rbac/src/main/java/com/azure/management/graphrbac/implementation/GraphRbacManager.java

```diff
@@ -124,14 +140,24 @@ private GraphRbacManager(RestClient restClient, String tenantId) {
                 .host(graphEndpoint)
                 .tenantID(tenantId)
                 .build();
+        if (subscriptionId == null) {
+            subscriptionId = subscriptionIdFromRestClient(restClient);
+        }
         this.authorizationManagementClient = new AuthorizationManagementClientBuilder()
                 .pipeline(restClient.getHttpPipeline())
                 .host(resourceManagerEndpoint)
-                .subscriptionId(((AzureTokenCredential) restClient.getCredential()).getDefaultSubscriptionId())
+                .subscriptionId(subscriptionId)
                 .build();
         this.tenantId = tenantId;
     }
 
+    private String subscriptionIdFromRestClient(RestClient restClient) {
+        if (restClient.getCredential() != null && restClient.getCredential() instanceof AzureTokenCredential) {
+            return ((AzureTokenCredential) restClient.getCredential()).getDefaultSubscriptionId();
+        }
+        throw new IllegalArgumentException("The credential in RestClient must be an instance of AzureTokenCredential.");
```

Comment:
The AzureTokenCredential's default subs is null, I think return null in subs may be better than throw exception.

Reply:
The question is whether we should inform user about this cast failure. If not mandatory, we can just return null instead of throwing exception.

Reply:
If a user uses restclient directly rather than the credential in authentication, I think they already know what they want.